akka-stream-tests/src/test/java/akka/stream/javadsl/FlowTest.java

```diff
@@ -682,6 +683,18 @@ public void apply(String elem) {
     probe.expectMsgEquals("C");
   }
 
+  @Test
+  public void mustBeAbleToUseCollectType() throws Exception{
+    final TestKit probe = new TestKit(system);
+    final Iterable<FlowSpec.Apple> input = Collections.singletonList(new FlowSpec.Apple());
+    final Flow<FlowSpec.Apple,FlowSpec.Fruit,NotUsed> fruitFlow = Flow.of(FlowSpec.Apple.class)
+            .collectType(FlowSpec.Fruit.class);
+    Source.from(input).via(fruitFlow.collectType(FlowSpec.Apple.class).collectType(FlowSpec.Fruit.class))
+            .collectType(FlowSpec.Apple.class)
+            .runForeach((apple) -> probe.getRef().tell(apple,ActorRef.noSender()),materializer);
+    probe.expectMsgAnyClassOf(FlowSpec.Apple.class);
```

Comment:
Just collecting to a `Sink.seq` and asserting on the materialized value from it would be cleaner I think, since there is no indeterminism going on here.

Reply:
Okay